### DISCUSSION TITLE
Why is Generic Access Profile (GAP) service hidden?

### DISCUSSION THREAD

**Discussion by marvinlehmann** [bluez/bluez#1052] [Q&A] [answered]
Hi!

I'm just wondering why the GAP service gets hidden in e.g. bluetoothctl or when using bluez through various libraries/tools (https://github.com/hbldh/bleak/issues/553#issuecomment-851006681, https://github.com/Jakeler/ble-serial/issues/110). I wanted to read the device name with [tinygo bluetooth](https://github.com/tinygo-org/bluetooth) but I can't access the GAP service with it either. 

The gatttool lists that service and can read the device's name just fine (`char-read-uuid 2a00`).

I also couldn't find the section in the bluez code where GAP (0x1800) gets excluded.

**Comment by Vudentz** [comment, marked as answer]
It is not hidden anymore: 

https://github.com/bluez/bluez/commit/dbd6591bd1d02ace39debd8a67c75b3cbe9c4d66

So by default this should be read-only, although we haven't made a release yet with these changes.
